testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpServerTest.java

```diff
@@ -0,0 +1,709 @@
+          assertThat(attrs).containsEntry(SemanticAttributes.HTTP_STATUS_CODE, endpoint.status);
+
+          assertThat(attrs)
+              .hasEntrySatisfying(
+                  SemanticAttributes.HTTP_FLAVOR, entry -> assertThat(entry).isIn("1.1", "2.0"));
+          assertThat(attrs).containsEntry(SemanticAttributes.HTTP_USER_AGENT, TEST_USER_AGENT);
+
+          assertThat(attrs).containsEntry(SemanticAttributes.HTTP_SCHEME, "http");
+          assertThat(attrs)
+              .hasEntrySatisfying(
+                  SemanticAttributes.HTTP_HOST,
+                  entry -> assertThat(entry).isIn("localhost", "localhost:" + port));
+          if (endpoint != INDEXED_CHILD) {
+            assertThat(attrs)
+                .containsEntry(
+                    SemanticAttributes.HTTP_TARGET,
+                    endpoint.resolvePath(address).getPath()
+                        + (endpoint == QUERY_PARAM ? "?" + endpoint.body : ""));
+          }
+
+          if (httpAttributes.contains(SemanticAttributes.HTTP_REQUEST_CONTENT_LENGTH)
+              || attrs.get(SemanticAttributes.HTTP_REQUEST_CONTENT_LENGTH) != null) {
+            assertThat(attrs)
+                .hasEntrySatisfying(
+                    SemanticAttributes.HTTP_REQUEST_CONTENT_LENGTH,
+                    entry -> assertThat(entry).isInstanceOf(Long.class));
+          }
+          if (httpAttributes.contains(SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH)
+              || attrs.get(SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH) != null) {
+            assertThat(attrs)
+                .hasEntrySatisfying(
+                    SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH,
+                    entry -> assertThat(entry).isInstanceOf(Long.class));
+          }
+          if (httpAttributes.contains(SemanticAttributes.HTTP_SERVER_NAME)) {
+            assertThat(attrs)
+                .hasEntrySatisfying(
+                    SemanticAttributes.HTTP_SERVER_NAME,
+                    entry -> assertThat(entry).isInstanceOf(String.class));
+          }
+          if (httpAttributes.contains(SemanticAttributes.HTTP_ROUTE)) {
+            String expectedRoute = options.expectedHttpRoute.apply(endpoint);
+            if (expectedRoute != null) {
+              assertThat(attrs).containsEntry(SemanticAttributes.HTTP_ROUTE, expectedRoute);
+            }
+          }
+
+          if (endpoint == CAPTURE_HEADERS) {
+            assertThat(attrs)
+                .containsEntry("http.request.header.x_test_request", new String[] {"test"});
+            assertThat(attrs)
+                .containsEntry("http.response.header.x_test_response", new String[] {"test"});
+          }
+          if (endpoint == CAPTURE_PARAMETERS) {
+            assertThat(attrs)
+                .containsEntry(
+                    "servlet.request.parameter.test_parameter", new String[] {"test value õäöü"});
+          }
+        });
+
+    return span;
+  }
+
+  private static void assertException(
```

Comment:
Can you use `SpanDataAssert.hasException`?

Reply:
I didn't use `SpanDataAssert.hasException` because it requires an instance of `Throwable` but here we only have exception class and error message. As actually we use only a few exception classes and all of them have a suitable constructor I could try constructing the exception instance reflectively.

Reply:
> we use only a few exception classes and all of them have a suitable constructor I could try constructing the exception instance reflectively

I think this approach makes the test harder to read, @anuraaga what do you think about something like `SpanDataAssert.hasExceptionSatisying` for the most flexibility here?

@laurit I'm ok merging with any approach here and we can revisit

Reply:
I think that could be useful, but in this case, can't the places passing an exception class and String error message pass a Throwable instead by calling `new Class(message)`?

Reply:
I think it should be done in a separate pr as this pr is already fairly large.